### PR TITLE
Make Lambda layer hash include transitive requirements (#2222)

### PR DESCRIPTION
**Pull request by jessebrennan**
Author (before primary review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `HCA` label <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes <sub>or this PR does not need porting</sub> 
- [x] Added `[r]` prefix at start of commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Ran `make requirements_update` <sub>or this PR does not change requirements*.txt</sub>
- [x] Mentioned `make requirements` in UPGRADING.rst <sub>or this PR does not change requirements*.txt</sub>
- [x] Documented upgrade of personal deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `[r]` prefix of commit title
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox` <sub>or commented that sandbox will be skipped</sub>
- [x] Reindexed `sandbox` <sub>or this PR does not require reindexing</sub>
- [x] Added PR reference to merge commit
- [x] Moved linked issue to Merged
- [x] Pushed merge commit to Github

Primary reviewer (after pushing merge commit)

- [ ] Moved freebies to Merged column <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab
- [ ] Reindexed `dev` <sub>or this PR does not require reindexing</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Unassign reviewer from PR

### CODE REVIEW

File: src/azul/lambda_layer.py [hunk 1]
```diff
@@ -82,6 +82,8 @@ def _build_package(self, input_zip, output_zip):
...
     @cached_property
     def object_key(self):
-        path = Path(config.project_root) / 'requirements.txt'
-        sha1 = file_sha1(path)
-        return f'{config.lambda_layer_key}/{sha1}.zip'
+        requirements = Path(config.project_root) / 'requirements.txt'
+        requirements_trans = Path(config.project_root) / 'requirements.trans.txt'
+        sha = file_sha1(requirements)
+        sha_trans = file_sha1(requirements_trans)
+        return f'{config.lambda_layer_key}/{sha}.{sha_trans}.zip'
```
Comment (achave11) [inline src/azul/lambda_layer.py:89]:
```suggestion
        sha = '.'.join(file_sha1(x) for x in (requirements, requirements_trans))
        return f'{config.lambda_layer_key}/{sha}.zip'
```
I wonder if consolidating the `sha1` operation to a single line is appeals to you? Let me know what you think.

**Review by achave11 — APPROVED** [review]
Looks good, just a suggestion where we could reduce the diff. but it's up to you. If you decide to go with it, not sure if the `sha` variable is descriptive enough.

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
Hard to demo.

**Comment by jessebrennan** [comment]
I can demo this by showing the file with the new, longer hash locally when
I make a change to the transitive requirements file.

**Comment by hannes-ucsc** [comment]
Sounds good.

**Pull request merged**